internal/storage/cheques.go

```
@@ -0,0 +1,223 @@
+}
+
+func (s *session) GetCheque(ctx context.Context, chequebookID string) (*models.Cheque, error) {
+	cheque := &cheque{}
+	if err := s.tx.StmtxContext(ctx, s.storage.getChequeByID).GetContext(ctx, cheque, chequebookID); err != nil {
+		if err != sql.ErrNoRows {
+			s.logger.Error(err)
+		}
+		return nil, upgradeError(err)
+	}
+	return s.storage.modelFromCheque(cheque)
+}
+
+func (s *session) AddCheque(ctx context.Context, cheque *models.Cheque) error {
+	result, err := s.tx.NamedStmtContext(ctx, s.storage.addCheque).
+		ExecContext(ctx, chequeFromModel(cheque))
+	if err != nil {
+		s.logger.Error(err)
+		return upgradeError(err)
+	}
+	if rowsAffected, err := result.RowsAffected(); err != nil {
+		s.logger.Error(err)
+		return upgradeError(err)
+	} else if rowsAffected != 1 {
+		return fmt.Errorf("failed to add cheque: expected to affect 1 row, but affected %d", rowsAffected)
+	}
+	return nil
+}
+
+func (s *session) UpdateCheque(ctx context.Context, cheque *models.Cheque) error {
+	result, err := s.tx.NamedStmtContext(ctx, s.storage.updateCheque).
+		ExecContext(ctx, chequeFromModel(cheque))
+	if err != nil {
+		s.logger.Error(err)
+		return upgradeError(err)
+	}
+	if rowsAffected, err := result.RowsAffected(); err != nil {
+		s.logger.Error(err)
+		return upgradeError(err)
+	} else if rowsAffected != 1 {
+		return fmt.Errorf("failed to update cheque: expected to affect 1 row, but affected %d", rowsAffected)
+	}
+	return nil
+}
+
+func (s *session) GetNotCashedCheques(ctx context.Context) ([]models.Cheque, error) {
+	cheques := []models.Cheque{}
+	rows, err := s.tx.StmtxContext(ctx, s.storage.getNotCashedCheques).QueryxContext(ctx)
+	if err != nil {
+		s.logger.Error(err)
+		return nil, upgradeError(err)
+	}
+	for rows.Next() {
+		cheque := &cheque{}
+		if err := rows.StructScan(cheque); err != nil {
+			s.logger.Error(err)
+			return nil, upgradeError(err)
+		}
+		model, err := s.storage.modelFromCheque(cheque)
+		if err != nil {
```

it would make sense to log the error and continue (skip a certain problematic cheque) instead of returning an error and abort the process, right? Otherwise a problematic cheque could sabotage the whole cashout routine

---

Makes sense, ty